packaging/installer/netdata-updater.sh

```diff
@@ -170,50 +167,44 @@ export NETDATA_LIB_DIR="${NETDATA_LIB_DIR:-${NETDATA_PREFIX}/var/lib/netdata}"
 [[ -z "${NETDATA_TARBALL_CHECKSUM}" ]] && [[ -f ${NETDATA_LIB_DIR}/netdata.tarball.checksum ]] && NETDATA_TARBALL_CHECKSUM="$(cat "${NETDATA_LIB_DIR}/netdata.tarball.checksum")"
 
 if [ "${INSTALL_UID}" != "$(id -u)" ]; then
-	fatal "You are running this script as user with uid $(id -u). We recommend to run this script as root (user with uid 0)"
+  fatal "You are running this script as user with uid $(id -u). We recommend to run this script as root (user with uid 0)"
 fi
 
 logfile=
 if [ -t 2 ]; then
-	# we are running on a terminal
-	# open fd 3 and send it to stderr
-	exec 3>&2
+  # we are running on a terminal
+  # open fd 3 and send it to stderr
+  exec 3>&2
 else
-	# we are headless
-	# create a temporary file for the log
-	logfile=$(mktemp ${logfile}/netdata-updater.log.XXXXXX)
-	# open fd 3 and send it to logfile
-	exec 3>"${logfile}"
+  # we are headless
+  # create a temporary file for the log
+  logfile=$(mktemp ${logfile}/netdata-updater.log.XXXXXX)
+  # open fd 3 and send it to logfile
+  exec 3> "${logfile}"
 fi
 
 set_tarball_urls "${RELEASE_CHANNEL}" "${IS_NETDATA_STATIC_BINARY}"
 
 if [ "${IS_NETDATA_STATIC_BINARY}" == "yes" ]; then
-	TMPDIR="$(create_tmp_directory)"
-	PREVDIR="$(pwd)"
-
-	echo >&2 "Entering ${TMPDIR}"
-	cd "${TMPDIR}"
-
-	download "${NETDATA_TARBALL_CHECKSUM_URL}" "${TMPDIR}/sha256sum.txt"
-	download "${NETDATA_TARBALL_URL}" "${TMPDIR}/netdata-latest.gz.run"
-	if ! grep netdata-latest.gz.run "${TMPDIR}/sha256sum.txt" | safe_sha256sum -c - >/dev/null 2>&1; then
-		fatal "Static binary checksum validation failed. Stopping netdata installation and leaving binary in ${TMPDIR}"
-	fi
-
-	# Do not pass any options other than the accept, for now
-	sh "${TMPDIR}/netdata-latest.gz.run" --accept ${REINSTALL_OPTIONS}
-
-	#shellcheck disable=SC2181
-	if [ $? -eq 0 ]; then
-		rm -r "${TMPDIR}"
-	else
-		echo >&2 "NOTE: did not remove: ${TMPDIR}"
-	fi
-	echo >&2 "Switching back to ${PREVDIR}"
-	cd "${PREVDIR}"
+  TMPDIR="$(create_tmp_directory)"
+  PREVDIR="$(pwd)"
+
+  echo >&2 "Entering ${TMPDIR}"
+  cd "${TMPDIR}" || exit 1
+
+  download "${NETDATA_TARBALL_CHECKSUM_URL}" "${TMPDIR}/sha256sum.txt"
+  download "${NETDATA_TARBALL_URL}" "${TMPDIR}/netdata-latest.gz.run"
+  if ! grep netdata-latest.gz.run "${TMPDIR}/sha256sum.txt" | safe_sha256sum -c - > /dev/null 2>&1; then
+    fatal "Static binary checksum validation failed. Stopping netdata installation and leaving binary in ${TMPDIR}"
+  fi
+
+  # Do not pass any options other than the accept, for now
+  if ! sh "${TMPDIR}/netdata-latest.gz.run" --accept "${REINSTALL_OPTIONS}"; then
```

Comment:
But this way, you never run `rm -r "${TMPDIR}"`

```suggestion
  if sh "${TMPDIR}/netdata-latest.gz.run" --accept "${REINSTALL_OPTIONS}"; then
    rm -r "${TMPDIR}"
  else
    echo >&2 "NOTE: did not remove: ${TMPDIR}"
  fi
```

Reply:
Done and done